runtime/query.go

```diff
@@ -20,6 +21,29 @@ type Query interface {
 }
 
 func (r *Runtime) Query(ctx context.Context, instanceID string, query Query, priority int) error {
-	// TODO: Add caching here
-	return query.Resolve(ctx, r, instanceID, priority)
+	// take all deps and their last updated time and add them to the query key, prefix query key with instanceID
+	cacheKey := instanceID + query.Key()
```

Comment:
can be added to make it human readable but not sure if we need it, I am fine with anything

Reply:
well, the actual need for some separators would be to make the cache keys unique when concatenating multiple names together, e.g. 
consider a boundary case where concatenating strings - (`abc` and `def`) or (`ab` an `cdef` ) without a separator both will yield same key. We can also consider an approach similar to CacheKeyBuilder in Druid by using a unique byte as type of entity to be appended as a separator.

Reply:
If we have query type string like (topk, nullcount etc.) in the query key, I highly doubt we will get in to this issue but anyways we can do it to be on safe side. CacheKeyBuilder approach can help in keeping key sizes in check, do you see any other benefit to add that complexity ?

Reply:
I agree there should be a reliable separator (and escaping of that separator in values) to prevent the boundary cases Nishant describes – I've seen this before in FoundationDB's [tuple layer](https://github.com/apple/foundationdb/blob/main/design/tuple.md).

However, in this case, I'm pretty sure native Go structs compare by field (when not a pointer), so we might just be able to do something like:

```go
type queryCacheKey struct {
  instanceID string
  key string
}

// ...

cacheKey := queryCacheKey{instanceID: instanceID, key: query.Key()}
```

Reply:
changed cache key to the struct